modules/tree/main/index.coffee

```diff
@@ -1,5 +1,35 @@
+createNodeView = (node, renderer, lazy) ->
+  node.view('.hx-tree-node')
+    .enter (d) -> @append(createTreeNode(d, renderer, lazy)).node()
+
+createChildren = (children, renderer, lazy) ->
+  children.map (child) -> createTreeNode(child, renderer, lazy)
+
+createTreeNode = (data, renderer, lazy) ->
+  treeNode = hx.detached('div').class('hx-tree-node')
+
+  nodeContent = hx.detached('div').attr('class', 'hx-tree-node-content')
+  renderer(nodeContent.node(), data)
+  if data.children? and data.children.length > 0
+    childContainer = hx.detached('div').class('hx-tree-node-children').style('display', 'none')
+    if lazy
+      childView = createNodeView(childContainer, renderer, lazy)
+      hx.component.register(treeNode.node(), {
+        renderChildren: -> childView.apply(data.children)
```

Comment:
I couldn't think of a more reliable way to not only detect whether a node has children to render and actually _do_ the render. This is used in `format` -> `renderLazyChildren`